app/featuresmodel.cpp

```diff
@@ -158,6 +158,12 @@ int FeaturesModel::rowCount( const QModelIndex &parent ) const
 
 QVariant FeaturesModel::featureTitle( const FeatureLayerPair &featurePair ) const
 {
+  if ( !featurePair.layer() || !featurePair.layer()->isValid() ) 
+  {
+    CoreUtils::log( QStringLiteral( "Features Model" ), QStringLiteral( "Received invalid feature layer pair!" ) );
+    return tr( "Unknown title" );
```

Comment:
When `title` is empty we return `pair.feature().id()`
Should we not be returning the same for this case too?

Reply:
Not sure, if the layer is invalid, I'd expect everything is broken 💥

Reply:
OK, let's leave it like this. It might help identifying the edge case when this happens.